### PR TITLE
Thirdmodule

### PR DESCRIPTION
**Pull request by javierbhai**
Multiple changes

- First approach to layout. Atomic design-first approach
- Styles added by using Scss, Scss-loader added
- Typescript and propTypes added, TS-loader
- Error Handler added
- JPG, PNG loader added

### CODE REVIEW

File: babel.config.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
           "useBuiltIns": "usage"
         }
       ],
-      "@babel/react"
+      "@babel/react",
+      "@babel/typescript"
```
Comment (igorxciv) [inline babel.config.json:11]: 👍

File: public/css/reset.css [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+  *{
+	margin: 0;
+	padding: 0;
+	list-style: none;
+	text-decoration: none;
+	border: none;
+	outline: none;
```
Comment (igorxciv) [inline public/css/reset.css:8]: cool. this is a good practice to reset the CSS. also, there is a pre-configured CSS for resetting all default styles: https://meyerweb.com/eric/tools/css/reset/

File: src/App/ErrorBoundary/ErrorBoundary.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    const { children } = props
+
+    const ErrorMessage = () => (
+        <h2>
+            We are sorry, The Roulette just stops :( 
+        </h2>
+    )
+
+    let isEverythingOK  = true
+    return <> { isEverythingOK ? children : <ErrorMessage />}</>
```
Comment (igorxciv) [inline src/App/ErrorBoundary/ErrorBoundary.tsx:17]: what about providing error value to the props?

```suggestion
export const ErrorBoundary: React.FC<Props> = (props: Props) => {
    const { children, error } = props

    const ErrorMessage = () => (
        <h2>
            We are sorry, The Roulette just stops :( 
        </h2>
    )
    return error ? <ErrorMessage /> : children;
```

File: src/App/ErrorBoundary/ErrorBoundary.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+export const ErrorBoundary: React.FC<Props> = (props: Props) => {
+    const { children } = props
+
+    const ErrorMessage = () => (
+        <h2>
+            We are sorry, The Roulette just stops :( 
+        </h2>
+    )
```
Comment (igorxciv) [inline src/App/ErrorBoundary/ErrorBoundary.tsx:14]: also, it's ok to assign a value to variable as a component:

```suggestion
    const errorMessage = (
        <h2>
            We are sorry, The Roulette just stops :( 
        </h2>
    )
```

and then use it like this in return statement: 
```
return error ? errorMessage : children
```

File: src/pages/home/home.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { MainBanner } from "../../patterns/molecules/MainBanner";
+import { MovieList } from "../../patterns/molecules/MovieList";
+import { ErrorBoundary } from "../../App/ErrorBoundary";
+import { Footer } from "../../patterns/atoms/Footer";
```
Comment (igorxciv) [inline src/pages/home/home.tsx:5]: let's discuss these imports

File: src/patterns/atoms/Footer/Footer.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
+@import "../../theme/common.scss";
```
Comment (igorxciv) [inline src/patterns/atoms/Footer/Footer.scss:1]: I'd recommend to use absolute imports to design system. Let's discuss

File: src/patterns/atoms/MovieDate/MovieDate.tsx [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import React from "react"
+import "./MovieDate.scss"
```
Comment (igorxciv) [inline src/patterns/atoms/MovieDate/MovieDate.tsx:2]: lets try to add CSS Modules to make it more real-world application creation https://webpack.js.org/loaders/css-loader/#modules

We can discuss it during our session

File: src/patterns/atoms/Text/Text.tsx [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import React from "react"
+import "./Text.scss"
+type Props = {
+    children?: string,
+    size?: string
+}
+
+export const Text: React.FC<Props> = (props: Props) => {
+    const { children, size = 14 } = props
+    return <p className="text" style={{fontSize:`${size}px`}} >{children}</p>
```
Comment (igorxciv) [inline src/patterns/atoms/Text/Text.tsx:10]: it's not a good practice to add styles for sizing this way. let's discuss this on our Q&A session

File: src/patterns/molecules/MainBanner/MainBanner.scss [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        button{
+            background-color: var(--c-red);
+            border-radius: 5px;
+            color: var(--c-white);
+            font-family: $montserrat;
+            font-size: 20px;
+            width: 20%;
+        }
+    }
+    section{
```
Comment (igorxciv) [inline src/patterns/molecules/MainBanner/MainBanner.scss:27]: I'd recommend using the BEM methodology for organizing CSS selectors. Lets discuss

File: src/patterns/molecules/MovieList/MovieList.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+            description:"Action & Adventure",
+            date: "2014",
+            id: "r43f3"
+        },
+        {
+            title: "Avengers: War of Infinity",
+            description:"Action & Adventure",
+            date: "2014",
+            id: "r43t4"
+        }
```
Comment (igorxciv) [inline src/patterns/molecules/MovieList/MovieList.tsx:36]: let's move it to a separate JSON file

**Review by igorxciv — CHANGES_REQUESTED** [review]

File: src/patterns/atoms/MovieDate/MovieDate.tsx (hunk 7 above)
Comment (igorxciv) [reply src/patterns/atoms/MovieDate/MovieDate.tsx:2]: (not required, but nice to have and practice)

File: src/patterns/molecules/MovieList/MovieList.tsx (hunk 10 above)
Comment (igorxciv) [reply src/patterns/molecules/MovieList/MovieList.tsx:36]: (not required, but if you'd like to)

File: src/patterns/atoms/Footer/Footer.scss (hunk 6 above)
Comment (javierbhai) [reply src/patterns/atoms/Footer/Footer.scss:1]: Wooooooow this is just amazing
includePaths:[path.resolve(__dirname, 'src/patterns/')]

solved thanks

File: public/css/reset.css (hunk 2 above)
Comment (javierbhai) [reply public/css/reset.css:8]: Implemented 🚀

File: src/App/ErrorBoundary/ErrorBoundary.tsx (hunk 3 above)
Comment (javierbhai) [reply src/App/ErrorBoundary/ErrorBoundary.tsx:17]: implemented 🚀

File: src/App/ErrorBoundary/ErrorBoundary.tsx (hunk 4 above)
Comment (javierbhai) [reply src/App/ErrorBoundary/ErrorBoundary.tsx:14]: implemented 🚀

File: src/pages/home/home.tsx (hunk 5 above)
Comment (javierbhai) [reply src/pages/home/home.tsx:5]: implemented 🚀

File: src/patterns/atoms/Footer/Footer.scss (hunk 6 above)
Comment (javierbhai) [reply src/patterns/atoms/Footer/Footer.scss:1]: implemented <3

File: src/patterns/atoms/MovieDate/MovieDate.tsx (hunk 7 above)
Comment (javierbhai) [reply src/patterns/atoms/MovieDate/MovieDate.tsx:2]: :( not implemented jet I will

File: src/patterns/atoms/Text/Text.tsx (hunk 8 above)
Comment (javierbhai) [reply src/patterns/atoms/Text/Text.tsx:10]: implemented 🚀

File: src/patterns/molecules/MainBanner/MainBanner.scss (hunk 9 above)
Comment (javierbhai) [reply src/patterns/molecules/MainBanner/MainBanner.scss:27]: implemented 🚀

**Review by igorxciv — APPROVED** [review]

**Pull request merged**